src/main/java/org/wildfly/client/config/ConfigurationXMLStreamReader.java

```diff
@@ -725,13 +726,57 @@ default Expression getExpressionAttributeValue(int index, Expression.Flag... fla
         final String attributeValue = getAttributeValue(index);
         if (attributeValue == null) {
             return null;
+        } else if (attributeValue.startsWith("env.")) {
+            String envVar = System.getenv().get(attributeValue.substring(4));
+
+            if (envVar.startsWith("ENC:")) {
+                return resolveEncryptedExpression(envVar, flags);
+            } else {
+                return Expression.compile(envVar, flags);
+            }
+        } else if (attributeValue.contains("env") || attributeValue.contains("ENV")) {
+            String envVar = replaceNonAlphanumericByUnderscoresAndMakeUpperCase(attributeValue);
+
+            if (envVar.startsWith("ENC:")) {
+                return resolveEncryptedExpression(envVar, flags);
+            } else {
+                return Expression.compile(envVar, flags);
+            }
+        } else if (attributeValue.startsWith("prop.")) {
+            String propertyValue = System.getProperty(attributeValue.substring(5));
+
+            if (propertyValue.startsWith("ENC:")) {
+                return resolveEncryptedExpression(propertyValue, flags);
+            } else {
+                return Expression.compile(propertyValue, flags);
+            }
+        } else if (attributeValue.contains("prop") || attributeValue.contains("PROP")) {
+            String propertyValue = System.getProperty(replaceNonAlphanumericByUnderscoresAndMakeUpperCase(attributeValue));
+
+            if (propertyValue.startsWith("ENC:")) {
+                return resolveEncryptedExpression(propertyValue, flags);
+            } else {
+                return Expression.compile(propertyValue, flags);
+            }
+        } else if (attributeValue.contains("ENC:")) {
+            return resolveEncryptedExpression(attributeValue, flags);
         } else try {
             return Expression.compile(attributeValue, flags);
         } catch (IllegalArgumentException ex) {
             throw msg.expressionParseException(ex, getAttributeName(index), getLocation());
         }
     }
 
+    default Expression resolveEncryptedExpression(String attributeValue,  Expression.Flag... flags) throws ConfigXMLParseException {
+        Iterator resolverProviderIterator = ServiceLoader.load(ResolverProvider.class).iterator();
```

Comment:
What class loader is expected to be used for loading the provider?

Reply:
I am not exactly sure. 
Here is the [implementation of the ResolverClass](https://github.com/wildfly-security/wildfly-elytron/pull/2084/files#diff-26ddef11ba5f01d385a8e42995abfc408764299ab18c6c596d8e1c82da6502a5)
And [this](https://github.com/wildfly-security/wildfly-elytron/pull/2084/files#diff-a5b1c677a7799031ba4fc7e6e0baed779b2a6c268ac8ca75c6f3c711dfcc7f8f) is the class that we are trying to access using the service loader.

Reply:
We could consider supporting multiple if we can detect if a resolver supported a specific expression format or not so if a resolver did not resolve the expression the next one can be tried.

Reply:
I think then that the appropriate thing would be to load from its own class loader; e.g. `ServiceLoader.load(ResovlerProvider.class, ConfigurationXMLStreamReader.class.getClassLoader())`.

Be sure to wrap the whole interaction with the iterator - including both the `hasNext` and `next` calls - in a `try`/`catch` to catch `ServiceConfigurationError`. You can either skip it (if it is OK to try another provider), or wrap & throw it (if there is a sole provider which is expected to work).

Reply:
I have made the updates. Thanks for your review! Please feel free to let me know if there are any other changes I should made. 

Also, should I add a new WFCC issue? Or is the WFLY issue okay? I have a separate ELY issue related to this RFE, so I can do the same for WFCC.